frontend-react/src/components/FileHandlers/FileHandlerMessaging.tsx

```diff
@@ -220,21 +209,115 @@ export const FileWarningBanner = ({ message }: FileWarningBannerProps) => {
     );
 };
 
+const HL7_PRODUCT_MATRIX_URL =
+    "https://www.hl7.org/implement/standards/product_brief.cfm";
+const CDC_LIVD_CODES_URL = "https://www.cdc.gov/csels/dls/livd-codes.html";
+
+export type ValidationErrorMessageProps = {
+    errorCode: ErrorCode;
+    field?: string;
+    message?: string;
+};
+
+export function ValidationErrorMessage({
+    errorCode,
+    field,
+    message,
+}: ValidationErrorMessageProps) {
+    let child: ReactNode;
+
+    switch (errorCode) {
+        case ErrorCode.INVALID_MSG_PARSE_BLANK:
+            child = (
+                <>
+                    Blank message(s) found within file. Blank messages cannot be
+                    processed.
+                </>
+            );
+            break;
+        case ErrorCode.INVALID_HL7_MSG_TYPE_MISSING:
+            child = (
+                <>
+                    Missing required HL7 message type field MSH-9. Fill in the
+                    blank field before resubmitting.
+                </>
+            );
+            break;
+        case ErrorCode.INVALID_HL7_MSG_TYPE_UNSUPPORTED:
+            child = (
+                <>
+                    We found an unsupported HL7 message type. Please reformat to
+                    ORU-RO1. Refer to{" "}
+                    <USExtLink href={HL7_PRODUCT_MATRIX_URL}>
+                        HL7 specification
+                    </USExtLink>{" "}
+                    for more details.
+                </>
+            );
+            break;
+        case ErrorCode.INVALID_HL7_MSG_FORMAT_INVALID:
+            child = (
+                <>
+                    Invalid HL7 message format. Check your formatting by
+                    referring to{" "}
+                    <USExtLink href={HL7_PRODUCT_MATRIX_URL}>
+                        HL7 specification
+                    </USExtLink>
+                    .
+                </>
+            );
+            break;
+        case ErrorCode.INVALID_MSG_PARSE_DATETIME:
+            child = <>Reformat {field} as YYYYMMDDHHMM[SS[.S[S[S[S]+/-ZZZZ.</>;
+            break;
+        case ErrorCode.INVALID_MSG_PARSE_TELEPHONE:
+            child = (
+                <>
+                    Reformat phone number to a 10-digit phone number (e.g. (555)
+                    555-5555).
+                </>
+            );
+            break;
+        case ErrorCode.INVALID_HL7_MSG_VALIDATION:
```

Comment:
Looks like this description is duplicated -- I'm following up with Audrey to figure this out.

Reply:
I think INVALID_HL7_MSG_VALIDATION maps to row #2 in the .xls doc and if so, we will need a different message since we do not get $terserSpec: ${e.message} returned from the backend.